src/plugins/PageGenerator.cpp

```diff
@@ -2,20 +2,71 @@
+		("The inventor of the Pringles can is now buried in one.")
+		("According to British law, any unclaimed swan swimming in the open waters of England and Wales belongs to the Queen.")
+		("You are 13.8 percent more likely to die on your birthday.")
+		("Mulan has the highest kill-count of any Disney character.")
+		("You can eat and drink anything at least once.")
+		("What's red and bad for your teeth? A brick.")
+		("An apple a day keeps the doctor away. Or at least it does if you throw it hard enough.")
+		("My son, who's into astronomy, asked me how stars die. \"Usually an overdose, son,\" I told him.")
+		("If you donate one kidney, everybody loves you, and you're a total hero. But donate five and suddenly everyone is yelling. Sheesh!")
+		("Banging your head against a wall uses 150 calories an hour.")
+		("Chuck Norris counted to infinity, twice.")
+		("A group of frogs is called an army.")
+		;
+
 PageGenerator::PageGenerator() : Plugin("page_404_gen") { }
 
 PageGenerator::~PageGenerator() { }
 
 bool PageGenerator::onHandleError(NotApache::HTTPClient &client, int code) {
 	if (code == 404) {
+		static int x;
+		if (!x) {
+			std::srand(time(0)); // set seed only once
+			x++;
+		}
 		NotApache::globalLogger.logItem(logger::DEBUG, "Page Generator running");
-		client.data.response.setResponse(
-			client.data.response.builder.setStatus(code).setBody("Random text").build()
+		client.data.response.builder
+			// set response code
+			.setStatus(code)
+			// set mime type
+			.setHeader("Content-Type", "text/html")
+			// generate html code
+			.setBody(
+				std::string("<!DOCTYPE html><html><head> <title>404 Not Found</title> <style>")
+				+ std::string("*{ transition:all 0.6s; }")
+				+ std::string("html{ height:100%; background-image:url(\"https://picsum.photos/1920/1080\"); background-color:#E0E0E0; background-repeat:no-repeat; background-size:cover; }")
+				+ std::string("body{ font-family:'Lato', sans-serif; color:#575656; margin:0; }")
+				+ std::string("#main{ display:table; width:100%; height:100vh; text-align:center; }")
+				+ std::string(".fof{ display:table-cell; vertical-align:middle; }")
+				+ std::string(".box{ background-color:#E0E0E0; padding-bottom:20px; margin-left:350px; margin-right:350px; }")
+				+ std::string(".fof h1{ font-size:80px; display:inline-block; padding-right:12px; animation: type .5s alternate infinite; }")
+				+ std::string(".fof p{ font-size:20px; }")
+				+ std::string("@keyframes type{ from{ box-shadow: inset -3px 0px 0px #575656; } to{ box-shadow:inset -3px 0px 0px transparent; } }")
+				+ std::string("</style></head><body>")
+				+ std::string("<div id=\"main\"><div class=\"fof\"><div class=\"box\"><br>")
+				+ std::string("<h1>Error 404</h1><br>")
+				+ std::string("<p>") + _facts[std::rand() % _facts.size()] + "</p>"  // get a random fact
+				+ std::string("</div></div></div></body></html>")
```

Comment:
Use a `static const std::string` for the template and when you need to generate, just copy if, find something like `{{FACT}}` and replace it with your fact. (I did something similar in `ConfigException::PrettyPrint()`, you can look at that as example)